### PR TITLE
[4.0] Drop ReCaptcha v1 support, migrate to Google ReCaptcha library

### PR DESCRIPTION
**Pull request by mbabker**
Pull Request for Issue #18503

### Summary of Changes

- Removes support for ReCaptcha v1
- Removes the (customized) `recaptchalib.php` file and migrates to Google's PHP ReCaptcha library
- Adds a bridge class implementing their `RequestMethod` interface to use our HTTP adapter
- Makes all library classes available in global libraries (bridge class and the Google library), should make it less required for extension devs to ship these resources if they choose not to use the core captcha implementation

### Testing Instructions

- ReCaptcha just works

### Documentation Changes Required

- Note removed v1 support
- Note new library support

### CODE REVIEW

**Comment by brianteeman** [comment]
The language file needs updating as well to remove the 1.0 strings

**Comment by mbabker** [comment]
> The language file needs updating as well to remove the 1.0 strings

Done

File: libraries/vendor/composer/installed.json [hunk 1]
```diff
@@ -2072,5 +2072,52 @@
...
+        "autoload": {
+            "psr-4": {
+                "ReCaptcha\\": "src/ReCaptcha"
+            }
+        },
+        "notification-url": "https://packagist.org/downloads/",
+        "license": [
+            "BSD-3-Clause"
+        ],
+        "description": "Client library for reCAPTCHA, a free service that protect websites from spam and abuse.",
```
Comment (brianteeman) [inline libraries/vendor/composer/installed.json:2114]: protects not protect

Comment (zero-24) [reply libraries/vendor/composer/installed.json:2114]: hmm isn't this a upstream issue with the package than? IIRC we don't generate the composer description our self 😄

Comment (brianteeman) [reply libraries/vendor/composer/installed.json:2114]: i dont know - i did search the upstream repo for the string first but didnt get a match - maybe i searched wrongly

Comment (zero-24) [reply libraries/vendor/composer/installed.json:2114]: https://github.com/google/recaptcha/blob/master/composer.json#L3 ;)

Comment (zero-24) [reply libraries/vendor/composer/installed.json:2114]: Do you want me to send a PR or do you want to do it yourself?

Comment (brianteeman) [reply libraries/vendor/composer/installed.json:2114]: Stupid search on github does not match partial words

pr submitted upstream

**Comment by wilsonge** [comment]
Looks good. Can you fix conflict please and we'll get this merged in :)

**Comment by mbabker** [comment]
Done.

**Comment by zero-24** [comment]
I have just added the `Documentation required` label so at some time we remember to document the changes made here.

**Pull request merged**